PathKit/Sources/PathKit/Line.swift

```diff
@@ -7,33 +7,30 @@
 
 import Foundation
 
-public struct LineSegment {
-    /// The starting point of `self`.
-    public var start: Point
+/// A line in 2-dimensional space represented in its normal form.
+public struct Line {
+    /// The inclination of the line in radians.
+    public var 𝜭: Double
 
-    /// The endpoint of `self`.
-    public var end: Point
+    /// The length of the normal segment.
+    /// The normal segment joins the origin with the closest point on the line to the origin.
+    public var p: Double
 
-    /// The Euclidean distance from `start` to `end`.
-    public var length: Double {
-        return distance(start, end)
+    /// Determines whether `point` is on the line.
+    public func contains(_ point: Point) -> Bool {
+        return (point.y * sin(𝜭) + point.x * cos(𝜭)) == p
     }
 
-    /// Evaluates the line at the given parameter value `t`.
-    ///
-    /// - Parameter t: A value between 0 and 1.
-    /// - Returns: The point
-    public func point(at t: Double) -> Point {
-        return lerp(start, end, at: t)
+    /// Determines if the two lines intersect.
+    public static func intersect(lhs: Line, rhs: Line) -> Bool {
+        // TODO: Deal with edge cases
+        return  lhs.𝜭 != rhs.𝜭 || lhs.p == rhs.p
```

Comment:
You need to normalize the angles (0 is the same as 2π). This can be avoided with an Angle type.

Reply:
Yeah those are one of the cases I had in mind.